rust/agama-dbus-server/src/network/nm/dbus.rs

```diff
@@ -159,6 +173,48 @@ fn wireless_config_to_dbus(conn: &WirelessConnection) -> NestedHash {
     ])
 }
 
+/// Converts a MatchConfig struct into a HashMap that can be sent over D-Bus.
+///
+/// * `match_config`: MatchConfig to convert.
+fn match_config_to_dbus(match_config: &MatchConfig) -> HashMap<&str, zvariant::Value> {
+    let mut match_config_dbus = HashMap::new();
+
+    let drivers: Value = match_config
+        .driver
+        .iter()
+        .map(|dr| dr.to_string())
```

Comment:
You intention here is to `clone` the strings, right? You can use the [cloned](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.cloned) function:

```suggestion
        .cloned()
```